doc/src/modules/physics/mechanics/symsystem.rst

```diff
@@ -0,0 +1,184 @@
+    >>> import sympy.physics.mechanics.system as system
+
+The first step will be to initialize all of the dynamic and constant symbols. ::
+
+    >>> x, y, u, v, lam = dynamicsymbols('x y u v lambda')
+    >>> m, l, g = symbols('m l g')
+
+Next step is to define the equations of motion in multiple forms:
+
+    [1] Explicit form where the kinematics and dynamics are combined
+        x' = F_1(x, t, r, p)
+
+    [2] Implicit form where the kinematics and dynamics are combined
+        M(x, p) x' = F_2(x, t, r, p)
+
+    [3] Implicit form where the kinematics and dynamics are separate
+        M(q, p) u' = F_3(q, u, t, r, p)
+        q' = G(q, u, t, r, p)
+
+where
+
+x : states, i.e. [q, u]
+t : time
+r : specified (exogenous) inputs
+p : constants
+q : generalized coordinates
+u : generalized speeds
+M : mass matrix or generalized inertia matrix (implicit dynamical equations or
+combined dynamics and kinematics)
+F : right hand side (implicit dynamical equations or combined dynamics and
+kinematics)
+G : right hand side of the kinematical differential equations ::
+
+    >>> dyn_implicit_mat = Matrix([[1, 0, -x/m],
+    ...                            [0, 1, -y/m],
+    ...                            [0, 0, l**2/m]])
+    >>> dyn_implicit_rhs = Matrix([0, 0, u**2 + v**2 - g*y])
+    >>> comb_implicit_mat = Matrix([[1, 0, 0, 0, 0],
+    ...                             [0, 1, 0, 0, 0],
+    ...                             [0, 0, 1, 0, -x/m],
+    ...                             [0, 0, 0, 1, -y/m],
+    ...                             [0, 0, 0, 0, l**2/m]])
+    >>> comb_implicit_rhs = Matrix([u, v, 0, 0, u**2 + v**2 - g*y])
+    >>> kin_explicit_rhs = Matrix([u, v])
+    >>> comb_explicit_rhs = comb_implicit_mat.LUsolve(comb_implicit_rhs)
+
+Now the reference frames, points and particles will be set up so this
+information can be passed into `system.SymbolicSystem` in the form of a bodies
+and loads iterable. ::
+
+    >>> theta = atan(x/y)
+    >>> N = ReferenceFrame('N')
+    >>> A = N.orientnew('A', 'Axis', [theta, N.z])
+    >>> O = Point('O')
+    >>> P = O.locatenew('P', l * A.x)
+    >>> Pa = Particle('Pa', P, m)
+
+Now the bodies and loads iterables need to be initialized. ::
+
+    >>> bodies = [Pa]
+    >>> loads = [(P, g * m * N.x)]
+
+The equations of motion are in the form of a differential algebraic equation
+(DAE) and DAE solvers need to know which of the equations are the algebraic
+expressions. This information is passed into `SymbolicSystem` as a list
+specifying which rows are the algebraic equations. In this example it is a
+different row based on the chosen equations of motion format. The row index
+should always correspond to the mass matrix that is being input to the
+`SymbolicSystem` class but will always correspond to the row index of the
+combined dynamics and kinematics when being accessed from the `SymbolicSystem`
+class. ::
+
+    >>> alg_con = [2]
+    >>> alg_con_full = [4]
```

Comment:
Do you have any idea if we could just check the equations internally? Is it easy to check if a given expression is algebraic or not? If so, it'd be nice to avoid this, especially since it is a little confusing wrt to input/output differences.